src/org/labkey/test/components/ui/entities/EntityInsertPanel.java

```diff
@@ -196,6 +234,44 @@ protected boolean isVisible(Locator locator)
         }
     }
 
+    private EntityInsertPanel showGrid()
+    {
+        if (!isGridVisible())
+            elementCache().modeSelectListItem("from Grid")
+                    .waitForElement(this, 2000).click();
+
+        WebDriverWrapper.waitFor(()-> isGridVisible(),
+                "the grid did bot become visible", 2000);
+        return this;
+    }
+
+    private EntityInsertPanel showFileUpload()
+    {
+        if (!isFileUploadVisible())
+            elementCache().modeSelectListItem("from File")
+                    .waitForElement(this, 2000).click();
+
+        WebDriverWrapper.waitFor(()-> isFileUploadVisible(),
+                "the file upload panel did bot become visible", 2000);
+        return this;
+    }
+
+    private void waitForLoaded()
+    {
+        WebDriverWrapper.waitFor(()-> isGridVisible() || isFileUploadVisible(),
+                "The insert panel did not become loaded", WAIT_FOR_JAVASCRIPT);
```

Comment:
This is part of the danger of treating this as a nested panel. It assumes that this starts on the grid and then when calling isFileUploadVisible the view will change by clicking a tab. Won't this by default then leave you on the file upload panel?

Reply:
the file upload container component is a child of the file upload pane, which is a child of the EntityInsertPanel component
![image](https://user-images.githubusercontent.com/16809856/110978295-0ae27880-8318-11eb-816e-ab3db7dfba55.png)

Reply:
My thought is that our component wrapper classes shouldn't require a page to do things that the component does.